h2/src/main/org/h2/command/Parser.java

```diff
@@ -2198,17 +2198,24 @@ private Prepared parseComment() {
             // if the db name is equal to the schema name
             ArrayList<String> list = Utils.newSmallArrayList();
             do {
-                list.add(readUniqueIdentifier());
+                if (currentTokenType != DOT) {
+                    list.add(readUniqueIdentifier());
+                } else if(database.getMode().allowEmptySchemaValuesAsDefaultSchema) {
```

Comment:
```SQL
CREATE TABLE TEST(ID INT);

COMMENT ON COLUMN ......PUBLIC.......TEST......ID IS 'comment';
```
is not rejected with default settings now.

Reply:
That happens in MSSQLServer mode, yes? In other modes the behaviour should not differ.

Reply:
No, it happens in Regular mode. You broke both methods in the similar way due to incorrect conditional code.

Reply:
`Parser` should not read and ignore unexpected tokens. For example, in this method you can't use `readIf(DOT)` when `.` is not valid for current parsing position.

Reply:
fixed see TestIgnoreCatalogs.canCommentOn()

Reply:
... ah that what missleading for me. I that readIf is invariant, if the token was not encountered. It fills the expectedlist, if the token was not found. I should have looked into the code. That concerns the other part I changed as well. So the check for .. must be done


> if (currentTokenType == DOT) {
>    read();
> }

Reply:
SQL Server does not have `COMMENT ON` command. It's simpler to remove changes for `..` here. It was requested to support only the wrong catalog here for consistency, because this setting in available in Regular mode too where `COMMENT ON` is a valid command.

However, you created the problem with `..` in another method and it isn't fixed yet.

Reply:
BTW, do you really need `..`?

Reply:
Sadly sqls-server seems not to support columns on from version 7 onward. sybase does, so therefore I would leave the handling there. "Sadly" the dbms we are using is sybase.... At another place I commented, that the construct is used. But admittedly not in "comment on". 
So if you insist, I can remove all the changes i did for "comment on" but I would not recommend it.
The changes I did last should fix the differences encountered in the not mssql-case.